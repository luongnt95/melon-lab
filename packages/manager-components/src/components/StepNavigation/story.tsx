import { storiesOf } from '@storybook/react';
import React from 'react';
import StepNavigation from './';

const data = {
  steps: [
    {
      name: 'Step 1',
      isCompleted: true,
    },
    {
      name: 'Step 2',
      isActive: true,
    },
    {
      name: 'Step 2',
    },
  ],
};

storiesOf('Components|Step Navigation', module).add('Default', () => {
  return <StepNavigation {...data} />;
});
