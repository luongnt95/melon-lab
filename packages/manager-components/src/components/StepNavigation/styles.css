.step-navigation {
  display: table;
  display: flex;
  margin: 0;
  padding: 0;
  width: 100%;
  background-color: color-mod(var(--color-grey) l(95%));
  overflow: hidden;
  counter-reset: a;
}

.step-navigation__step {
  position: relative;
  display: table-cell;
  display: inline-flex;
  justify-content: center;
  flex: 1 0 auto;
  min-width: 1px;
  list-style-type: none;
  text-align: center;
}

.step-navigation__step:after,
.step-navigation__step:before {
  z-index: 1;
  content: '';
  display: inline-block;
  width: 0;
  height: 0;
  border-style: solid;
  border-color: transparent transparent transparent var(--color-white);
  position: absolute;
  top: -6px;
  right: -20px;
  border-width: 34px 0 34px 20px;
}

.step-navigation__step:after {
  right: -18px;
  border-left-color: color-mod(var(--color-grey) l(95%));
}

.step-navigation__step--active {
  background-color: var(--color-secondary);
  font-weight: var(--font-weight-bold);
}

.step-navigation__step--active:after {
  border-left-color: var(--color-secondary);
}

.step-navigation__step--active .step-navigation__counter {
  background-color: var(--color-secondary-dark);
  color: var(--color-white);
}

.step-navigation__step--active .step-navigation__text {
  color: var(--color-black);
  position: static;
}

.step-navigation__step--completed {
  background-color: color-mod(var(--color-profit) l(75%));
}

.step-navigation__step--completed:after {
  border-left-color: color-mod(var(--color-profit) l(75%));
}

.step-navigation__step--completed .step-navigation__counter {
  background-color: var(--color-profit);
  color: var(--color-white);
}

.step-navigation__step--completed .step-navigation__text {
  color: var(--color-black);
}

.step-navigation__step:first-child .step-navigation__wrap {
  padding-left: 10px;
}

.step-navigation__step:last-child .step-navigation__wrap {
  padding-right: 10px;
}

.step-navigation__wrap {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  line-height: 0;
  display: block;
  padding: var(--space-m) 10px var(--space-m) 30px;
  color: #5a6972;
  width: 100%;
}

.step-navigation__counter {
  font-weight: 700;
  line-height: 16px;
  font-size: 13px;
  display: inline-block;
  padding: 4px 0;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  background-color: #bfd0d9;
  vertical-align: middle;
}

.step-navigation__counter:before {
  counter-increment: a;
  content: counter(a);
}

.step-navigation__text {
  line-height: 20px;
  margin-left: 5px;
  vertical-align: middle;
  position: static;
}
