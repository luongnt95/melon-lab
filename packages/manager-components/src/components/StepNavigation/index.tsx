import classNames from 'classnames';
import React, { StatelessComponent } from 'react';

import styles from './styles.css';

export interface StepNavigationProps {
  steps: Array<{
    name: string;
  }>;
  activeStep: number;
}

export const StepNavigation: StatelessComponent<StepNavigationProps> = ({
  steps,
}) => {
  const activeStepClassName = step =>
    classNames('step-navigation__step', {
      'step-navigation__step--active': step.isActive,
      'step-navigation__step--completed': step.isCompleted,
    });

  return (
    <ul className="step-navigation" data-c-name="StepNavigation">
      <style jsx>{styles}</style>
      {steps &&
        steps.map((step, i) => (
          <li key={i} className={activeStepClassName(step)}>
            <span className="step-navigation__wrap">
              <span className="step-navigation__counter" />
              <span className="step-navigation__text">{step.name}</span>
            </span>
          </li>
        ))}
    </ul>
  );
};

export default StepNavigation;
