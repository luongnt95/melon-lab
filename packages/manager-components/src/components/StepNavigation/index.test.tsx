import React from 'react';
import StepNavigation from './index';

const data = {
  restore: () => null,
  mnemonic: 'lorem ipsum',
};

describe('StepNavigation', () => {
  const defaultElement = <StepNavigation {...data} />;
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(defaultElement);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
