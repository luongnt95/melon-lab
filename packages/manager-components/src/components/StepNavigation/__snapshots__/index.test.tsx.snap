// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepNavigation should render correctly 1`] = `
<ul
  className="jsx-__scopedHash step-navigation"
  data-c-name="StepNavigation"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
</ul>
`;
